projects/soft_999_Linux_Heller_fw_102/src/main.c

```diff
@@ -0,0 +1,51 @@
+// Library Initializations
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "delay.h"       // For real-time delays.
+#include "interrupt.h"   // For interrupting stuff
+#include "log.h"         // For printing stuff.
+#include "soft_timer.h"  // Software timers for scheduling future events.
+
+#include "gpio.h"  // General Purpose I/O control.
+#include "misc.h"  // Various helper functions/macros.
+
+typedef struct struct_counter {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counter;
+
+void counter_timer_callback(SoftTimerId timer_id, void *context) {
+  Counter *counter_storage = context;
+
+  uint8_t value = counter_storage->counter_a % 2;
+
+  counter_storage->counter_a++;
+  LOG_DEBUG("Counter A: %u\n", counter_storage->counter_a);
+
+  if (value == 0) {
```

Comment:
You probably don't need the `value` variable here. Just toss in `counter_storage->counter_a % 2` into the if statement